core/src/main/java/org/mule/runtime/core/privileged/exception/TemplateOnErrorHandler.java

```diff
@@ -76,6 +91,29 @@
   protected String errorType = null;
   protected ErrorTypeMatcher errorTypeMatcher = null;
 
+  private final Supplier<FluxSink<CoreEvent>> routingSink = new RoundRobinFluxSinkSupplier<>(getRuntime().availableProcessors(),
+                                                                                             new OnErrorHandlerFluxObjectFactory());
+
+  private final class OnErrorHandlerFluxObjectFactory implements Supplier<FluxSink<CoreEvent>> {
+
+    @Override
+    public FluxSink<CoreEvent> get() {
+      final FluxSinkRecorder<CoreEvent> sinkRef = new FluxSinkRecorder<>();
+      Flux<CoreEvent> onErrorFlux =
+          Flux.create(sinkRef)
+              .map(beforeRouting())
+              .concatMap(route())
```

Comment:
this is as bad as `flatMap`, we need to look further into this in a next iteration